core/services/headtracker/head_listener_test.go

```diff
@@ -62,6 +186,7 @@ func Test_HeadListener_ResubscribesIfWSClosed(t *testing.T) {
 
 	// Expect a call to unsubscribe on error
 	sub.On("Unsubscribe").Once().Run(func(_ mock.Arguments) {
+		close(headsCh)
```

Comment:
Does unsubscribing in geth also close this channel?

Reply:
I checked this by adding close() right after Unsubscribe and this resulted in panic. I concluded it does close the channel, even though I could not find a proof and the interface is not clear about that either.
I also saw your ticket with Eth project about this topic, if you have more details on that, please share.

Reply:
https://github.com/ethereum/go-ethereum/issues/24116

Reply:
Alright, makes sense. Does HeadListener _rely_ on this behaviour (the closing of the channel)?

Reply:
I think it does not depend on that behaviour. Assuming we are constantly reconnecting and the channel is not actually closed, then it will just be leaking resources. At the time of Unsubscribe the main loop of heads forwarding is quit, so it is not affected. That said I would only be worried about resources leaking and under normal circumstances it should not happen often (or ever) given a good connectivity.

Reply:
They should be cleaned up by the garbage collector, as long as we aren't leaking a goroutine referencing them also. Closing is really only necessary as a signal, and doesn't inform the GC.